src/background/index.ts

```diff
@@ -0,0 +1,68 @@
+import browser, { WebRequest } from 'webextension-polyfill'
+import type { Locale, MessageType } from '../types'
+
+declare const chrome: any
+let locale: Locale = null
+
+function updateLocale(value: Locale) {
+  if (value === locale) return
+  locale = value || null
+  browser.storage.local.set({ locale })
+  browser.browserAction.setBadgeText({ text: locale })
+}
+
+async function sendMessageToActiveTabs(message: MessageType) {
```

Comment:
Since we're filtering by current window, I assume this only sends a message to *one* tab?

Reply:
Basically yes, its only one tab but wanted to keep it generic